code/datums/ai/monkey/monkey_controller.dm

```diff
@@ -67,16 +75,18 @@ have ways of interacting with a specific mob and control it.
 
 	if(HAS_TRAIT(pawn, TRAIT_PACIFISM)) //Not a pacifist? lets try some combat behavior.
 		return
-	if(length(enemies) || blackboard[BB_MONKEY_AGRESSIVE]) //We have enemies or are pissed
-
-		var/mob/living/selected_enemy
 
+	var/mob/living/selected_enemy
+	if(length(enemies) || blackboard[BB_MONKEY_AGRESSIVE]) //We have enemies or are pissed
+		var/list/valids = list()
 		for(var/mob/living/possible_enemy in view(MONKEY_ENEMY_VISION, living_pawn))
 			if(possible_enemy == living_pawn || (!enemies[possible_enemy] && (!blackboard[BB_MONKEY_AGRESSIVE] || HAS_AI_CONTROLLER_TYPE(possible_enemy, /datum/ai_controller/monkey)))) //Are they an enemy? (And do we even care?)
 				continue
+			// Weighted list, so the closer they are the more likely they are to be chosen as the enemy
+			valids[possible_enemy] = CEILING(100 / (get_dist(living_pawn, possible_enemy) || 1), 1)
```

Comment:
Since you're already there, can you make them skip choosing people in crit? Cause currently if they choose someone who's downed they'll try to shove them down disposals (see like, 20 lines below this) instead of fighting an enemy who is still standing

Reply:
This is fine, sometimes dealing with an enemy who has been downed temporarily over another aggressive enemy seems like acceptable behavior. If an active enemy is closer to them than the downed enemy they're far far more likely to go after the nearby enemy anyway, so you'll very rarely get monkeys walking past active enemies to try disposaling a downed one. It happening rarely though is part of the fun.

Reply:
Yeah, I guess that's fine; since before they'd pick the _first_ possible enemy they'd kinda just freeze if their target was downed and they were unable to reach a disposal bin. But now that it's randomized this is probably not an issue anymore.